### PR TITLE
pkg/lv_drivers: initial commit

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

Supersedes #16944 which is based on v7, this PR is therefore based on #17681. 

### Testing procedure

```
TEST_KCONFIG=1 BOARD=native make -C tests/pkg_lvgl/ clean flash term -j
TEST_KCONFIG=0 BOARD=native make -C tests/pkg_lvgl/ clean flash term -j
TEST_KCONFIG=1 BOARD=native make -C tests/pkg_lvgl_touch/ clean flash term -j
TEST_KCONFIG=0 BOARD=native make -C tests/pkg_lvgl_touch/ clean flash term -j
```

### Issues/PRs references

Closes #16944

### CODE REVIEW

File: pkg/lvgl/doc.txt [hunk 1]
```diff
@@ -26,4 +26,20 @@ Example of command line for changing the max activity period to 5s:
 CFLAGS=-DCONFIG_LVGL_ACTIVITY_PERIOD=5000 make -C tests/pkg_lvgl
 ```
 
+### SDL Requirements
+
+To run lvgl inside a simulator (native), SDL is required. Install instructions
```
Comment (kaspar030) [inline pkg/lvgl/doc.txt:31]: maybe add that the 32bit version of it is needed.

**Review by kaspar030 — APPROVED** [review]
ACK.

**Comment by kaspar030** [comment]
please squash!

File: pkg/lv_drivers/doc.txt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+On debian/ubuntu this could be as simple as installing the `libsdl2-dev:i386` package.
+
+Using SDL requires more stack so in case you are using it add
+
+```
+CFLAGS += '-DTHREAD_STACKSIZE_MAIN=48*1024'
+```
+
+to your makefile. 48kB is enough for the test other uses may need more or may
```
Comment (kaspar030) [inline pkg/lv_drivers/doc.txt:30]:
```suggestion
to your makefile. 48kB is enough for the test, other uses may need more or may
```

**Comment by fjmolinas** [comment]
Unrelated CI failure, should I re-trigger asap or re-run afterward skipping ic build https://ci.riot-os.org/RIOT-OS/RIOT/17713/6bb9656344481e48ca867fdedf1556c8d71c9676/output/compile/tests/driver_tps6274x/esp8266-esp-12x:gnu.txt? @kaspar030

**Comment by kaspar030** [comment]
> Unrelated CI failure, should I re-trigger asap or re-run afterward skipping ic build https://ci.riot-os.org/RIOT-OS/RIOT/17713/6bb9656344481e48ca867fdedf1556c8d71c9676/output/compile/tests/driver_tps6274x/esp8266-esp-12x:gnu.txt? @kaspar030

I re-triggered with "skip compile tests". there was only the one unrelated build failure.

**Comment by chrysn** [comment]
Just saw this when pulling in master, where it first gave me an error on `#include </SDL.h>` (granted, I didn't have libsdl2-dev installed) and now gives me:

```
$ make all term BOARD=native -C tests/pkg_lvgl
make: Entering directory '/home/chrysn/git/RIOT/tests/pkg_lvgl'
Building application "tests_pkg_lvgl" for "native" with MCU "native".

"make" -C /home/chrysn/git/RIOT/pkg/lv_drivers/ 
"make" -C /home/chrysn/git/RIOT/build/pkg/lv_drivers/sdl -f /home/chrysn/git/RIOT/pkg/lv_drivers/Makefile.lv_drivers_module MODULE=lv_drivers_sdl SRC=sdl.c
In file included from /usr/include/SDL2/SDL_stdinc.h:31,
                 from /usr/include/SDL2/SDL_main.h:25,
                 from /usr/include/SDL2/SDL.h:32,
                 from /home/chrysn/git/RIOT/build/pkg/lv_drivers/sdl/sdl.c:53:
/usr/include/SDL2/SDL_config.h:4:10: fatal error: SDL2/_real_SDL_config.h: No such file or directory
    4 | #include <SDL2/_real_SDL_config.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [/home/chrysn/git/RIOT/Makefile.base:146: /home/chrysn/git/RIOT/tests/pkg_lvgl/bin/native/lv_drivers_sdl/sdl.o] Error 1
make[1]: *** [Makefile:18: lv_drivers_sdl] Error 2
make: *** [/home/chrysn/git/RIOT/tests/pkg_lvgl/../../Makefile.include:799: pkg-build] Error 2
make: Leaving directory '/home/chrysn/git/RIOT/tests/pkg_lvgl'
```

which appears to be due to mine sitting in `/usr/include/x86_64-linux-gnu/SDL2/_real_SDL_config.h` (from libsdl2-dev) ... and `libsdl2-dev:i386` on Debian appears to conflict with, like, the rest of my system.

Is there anything that could be added to make this journey smoother? Checking for the existence of `sdl2-config` might be a good start. On which distributions was this tested?

**Comment by fjmolinas** [comment]
> Is there anything that could be added to make this journey smoother? Checking for the existence of sdl2-config might be a good start. On which distributions was this tested?

`ubuntu-21.10` and `ubuntu 20.10` on my side.

**Comment by chrysn** [comment]
Aaah, took me a major while to track: I merely couldn't install `libsdl2-dev:i386` due to an unrelated package that could not be updated. Now that I've installed it, all peachy.

What this would need in terms of checks is to see whether the library `sdl2-config` is referring to is actually i386 -- alas, I haven't found a way to do that portably. For now, I'm leaving here (and for discovery) that if you get errors like

```
In file included from <command-line>:
.../RIOT/tests/pkg_lvgl/bin/native/riotbuild/riotbuild.h:13:26: fatal error: /SDL.h: No such file or directory
   13 | #define SDL_INCLUDE_PATH </SDL.h>
      |                          ^
compilation terminated.
```

you'll need to install `libsdl2-dev:i386` on Debian-based systems, and if you get

```
/usr/include/SDL2/SDL_config.h:4:10: fatal error: SDL2/_real_SDL_config.h: No such file or directory
    4 | #include <SDL2/_real_SDL_config.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

you do have `libsdl2-dev` installed but it's the 64-bit version that native can't use.

**Pull request merged**